### PR TITLE
nix.perl-bindings: add passthru.perlModule

### PR DESCRIPTION
**Pull request by volth**
... so it can be used in `perl.withPackages`

### CODE REVIEW

**Comment by volth** [comment]
https://github.com/NixOS/nixpkgs/pull/59979 is the demo why it is needed.
Without the ```passthru.perlModule```, nix.perl-bindings won't be added to the available perl libraries.

**Comment by volth** [comment]
There is an idea to move ```nix.perl-packages``` into ```perl.pkgs.nix``` (besides magically adding ```passthru```, it would allow to use Nix bindings with any version of ```perl``` (the would be ```perl528.pkgs.nix``` and ```perldevel.pkgs.nix```)

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -154,18 +158,15 @@ in rec {
 
   nix = nixStable;
 
-  nix1 = callPackage common rec {
+  nix1 = perl.pkgs.toPerlModule(callPackage common rec {
```
Comment (alyssais) [inline pkgs/tools/package-management/nix/default.nix]:
```suggestion
  nix1 = perl.pkgs.toPerlModule (callPackage common rec {
```

**Comment by volth** [comment]
This should be done tree-wide, I submitted only `nix1` and `rrdtool` (https://github.com/NixOS/nixpkgs/pull/59976) because I needed only them.

Although I do not like `perl.pkgs.toPerlModule` and using `passthru` to mark derivations which have Perl modules inside, it is modeled after Nixpkgs's Python infrastructure, so it is good to keep the pattern for homogeneity.

**Comment by alyssais** [comment]
> This should be done tree-wide, I submitted only `nix1` and `rrdtool`
> (https://github.com/NixOS/nixpkgs/pull/59976) because I needed only
> them.

What is your suggested path forward, then?  Do you think this PR should
be merged or not?

**Comment by volth** [comment]
> What is your suggested path forward, then? Do you think this PR should be merged or not?

I am for the tree-wide change, but it will likely be stalled which kills all the enthusiasm to do it.

The realistic plan is to wait for some number of user-reported issues ("rrdtool does not work in `perl.withPackages(p: [rrdtool])`", etc) and then introduce the tree-wide change as the fix for multiple issues.

**Comment by alyssais** [comment]
> I am for the tree-wide change, but it will likely be stalled which kills all the enthusiasm to do it.
>
> The realistic plan is to wait for some number of user-reported issues ("rrdtool does not work in `perl.withPackages(p: [rrdtool])`", etc) and then introduce the tree-wide change as the fix for multiple issues.

Okay, let me put this another way -- is there any reason to _not_ merge
this now?

**Comment by volth** [comment]
> Okay, let me put this another way -- is there any reason to _not_ merge this now?

Only leaving the work in half-done state.
And, it you want to merge something, `rrdtool` would be more useful than `nix1`

**Comment by FRidh** [comment]
> Although I do not like perl.pkgs.toPerlModule and using passthru to mark derivations which have Perl modules inside, it is modeled after Nixpkgs's Python infrastructure, so it is good to keep the pattern for homogeneity.

What part do you not like? To mark packages that have modules? In the case of Python it is done to break recursion when searching for Python libs that need to be added to an environment.

**Pull request closed without merging**